jina/peapods/runtimes/gateway/prefetch.py

```diff
@@ -157,3 +139,132 @@ async def prefetch_req(num_req, fetch_to):
                 # this list dries, clear it and feed it with on_recv_task
                 prefetch_task.clear()
                 prefetch_task = [j for j in onrecv_task]
+
+
+class ZmqPrefetchCaller(BasePrefetchCaller):
```

Comment:
Whats the reasoning to further split the BasePrefetchCaller into this ZmqPrefetchCaller? I think that is slightly misleading, because the ZmqPrefetchCaller actually can use either gRPC or ZMQ. 
So for me it would make more sense to have one abstract (BasePrefetchCaller) Prefetcher which can handle all Flow internal traffic (either zmq or GRPC), but has no functionality to communicate those results back to the outside world.

Then we can have concrete Http/Grpc/Websocket Prefetchers implementing the missing communication logic to the outside.
It might even be better to use compositon over inheritance here? So some generic BasePrefetchCaller is just used by some http/grpc/websocket server wrapping code? I think functionally speaking we mix two different things here: Comunication with the `Flow` and the `Client`